realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -515,6 +517,17 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
 
             updateAsyncQueriesTask = null;
         }
+
+        executeAsyncTransactionCallbacks();
+    }
+
+    void executeAsyncTransactionCallbacks() {
+        if (!asyncTransactionCallbacks.isEmpty()) {
+            for (Runnable transactionCallback : asyncTransactionCallbacks) {
+                realm.handler.post(transactionCallback);
+            }
+            asyncTransactionCallbacks.clear();
```

Comment:
Be careful about the `clear()` here. There is a race condition that the executor task just added something without running and gets cleared here.